src/main/java/spoon/support/compiler/VirtualFolder.java

```diff
@@ -46,10 +46,10 @@ public void addFolder(SpoonFolder o) {
 	@Override
 	public List<SpoonFile> getAllFiles() {
 		List<SpoonFile> result = new ArrayList<>();
-
+		Set<SpoonFile> distinctSet = new HashSet<>();
 		for (SpoonFile f : getFiles()) {
 			// we take care not to add a file that was already found in a folder
-			if (!result.contains(f)) {

```

Comment:
I would prefer not to have equals/hashcode in those classes, in order to not to maintain them in the future. And consequently, it seems betetr to avoid a hashset here.